File: AI/ml/Machine Learning A-Z Template Folder/Part 1 - Data Preprocessing/REGRESSION_MASTER.R
# Polynominal Regression Template

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Splitting the dataset into the Training and Test set >>> Lets not do that here, 10 records only
#install.packages('caTools')
#library(caTools)
#set.seed(123)
#split = sample.split(dataset$Purchased, SplitRatio = 0.8)
#training_set = subset(dataset, split == TRUE)
#test_set = subset(dataset, split == FALSE)

# Feature Scaling >>> no need
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

# Fitting Regression to the dataset
# Create your regressor here

# Single Predicting new result
y_pred = predict(regressor, data.frame(Level = 6.5))
summary(y_pred)

# Visualize Regression Model
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  # actual values
  geom_point(aes(x = dataset$Level , y = dataset$Salary),
             colour = "red") +
  # prediction
  geom_line(aes(x = dataset$Level , y = predict(regressor, newdata = dataset)), 
            colour = "blue") +
  ggtitle('Truth or Bluff (REGRESION MODEL)') +
  xlab('Level') +
  ylab('Salary')

# ALT: Visualize Regression Model HIGH RES
# install.packages('ggplot2')
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
  # actual values
  geom_point(aes(x=dataset$Level , y=dataset$Salary),
             colour = "red") +
  # prediction
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))), 
            colour = "blue") +
  ggtitle('Truth or Bluff (REGRESION MODEL)') +
  xlab('Level') +
  ylab('Salary')

File: AI/ml/Machine Learning A-Z Template Folder/Part 1 - Data Preprocessing/REGRESSION_MASTER.py
# GENERAL REGRESSION TEMPLATE

##################################
### check code     : cmd + i   ###
### run code       : cmd enter ###
### set working dir: Fn F5     ###
##################################

# Import libraries

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values # A MATRIX
y = dataset.iloc[:, 2].values  # A VECTOR

# Splitting the dataset into the Training set and Test set
#from sklearn.cross_validation import train_test_split
#X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

# Feature Scaling
#from sklearn.preprocessing import StandardScaler
#sc_X = StandardScaler()
#X_train = sc_X.fit_transform(X_train)
#X_test = sc_X.transform(X_test)"""

# Fitting Regression Model to the dataset
# Create your regressor here

# Predicting a new result with Polynominal Regression
y_pred = regressor.predict(6.5)

# Visialize the Regression results
plt.scatter(X, y, color='red')  # actual
plt.plot(X, regressor.predict(X), color='blue')  # predicted
plt.title('Truth or Bluff (REGRESSION MODEL)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# ALT: Visialize the Regression results HIGH RES
X_grid = np.arange(min(X), max(X), 0.1) 
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color='red')  # actual
plt.plot(X, regressor.predict(X_grid), color='blue')  # predicted
plt.title('Truth or Bluff (REGRESSION MODEL)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()